### PR TITLE
Update pruner progress after fast sync is done.

### PR DESCRIPTION
**Pull request by grao1991**
### Description
Also fixed a bug that we mistakenly used the SchemaBatch for ledger db to prune genesis in state merkle db.

### CODE REVIEW

**Review by msmouse — COMMENTED** [review]
Looks legit.. possible to add a test somewhere? 
@JoshLind  Off the top of your head, is there somewhere in the tests where we can see if after a fast sync the data availability announcement begins at the snapshot version instead of 0?

**Comment by JoshLind** [comment]
> @JoshLind Off the top of your head, is there somewhere in the tests where we can see if after a fast sync the data availability announcement begins at the snapshot version instead of 0?

Yeah, unfortunately, there's no easy way to test this on the state sync side. My only guesses would be: (i) a new smoke test that brings up a fullnode using fast sync and then pings the metrics port to make sure the appropriate value is set in the relevant storage metrics?; or (ii) some sort of test on the db side?

**Comment by msmouse** [comment]
@grao1991  Guess it's the best if we make a test from the DB size to cover the while fast sync related APIs -- state sync unit tests nowadays have the Db mocked out entirely. If that's too much work short term, how about syncing a node up locally to test? 

@JoshLind We should have a smoke-test to cover fast sync...

**Comment by grao1991** [comment]
> @grao1991 Guess it's the best if we make a test from the DB size to cover the while fast sync related APIs -- state sync unit tests nowadays have the Db mocked out entirely. If that's too much work short term, how about syncing a node up locally to test?
> 
> @JoshLind We should have a smoke-test to cover fast sync...

Manually verified on a local node.

File: storage/aptosdb/src/lib.rs [hunk 1]
```diff
@@ -1893,18 +1893,49 @@ impl DbWriter for AptosDB {
...
 
             // Delete the genesis transaction
-            StateMerklePruner::prune_genesis(self.state_merkle_db.clone(), &mut batch)?;
             LedgerPruner::prune_genesis(
                 self.ledger_db.clone(),
                 self.state_store.clone(),
                 &mut batch,
             )?;
 
+            self.ledger_pruner
```
Comment (areshand) [inline storage/aptosdb/src/lib.rs:1902]: Looking great. Just to understand the context and feel free to skip. If a node uses fast sync to a version, The pruner can only start from this version. This is no way to prune to an even earlier version?

**Review by areshand — APPROVED** [review]

File: storage/aptosdb/src/lib.rs (hunk 1 above)
Comment (grao1991) [reply storage/aptosdb/src/lib.rs:1902]: Currently we don't sync any data from pervious versions.

**Comment by JoshLind** [comment]
> @JoshLind We should have a smoke-test to cover fast sync...

There are a bunch of smoke tests to cover fast sync (5+) IIRC. The question is whether or not you can inspect the correct storage metrics once it fast syncs, right? i.e., you check if the pruner has correctly been set then, right?

File: storage/aptosdb/src/lib.rs [hunk 2]
```diff
@@ -1893,18 +1893,49 @@ impl DbWriter for AptosDB {
...
+            let mut state_merkle_batch = SchemaBatch::new();
+            StateMerklePruner::prune_genesis(
+                self.state_merkle_db.clone(),
+                &mut state_merkle_batch,
+            )?;
+
+            self.state_store
+                .state_pruner
+                .pruner()
+                .save_min_readable_version(version, &state_merkle_batch)?;
```
Comment (JoshLind) [inline storage/aptosdb/src/lib.rs:1915]: There's a lot of individual calls to `save_min_readable_version`. If we do this anywhere else, I'd expect a wrapper/utility function, e.g., `save_min_readable_version_on_all_pruners`?

**Comment by msmouse** [comment]
> whether or not you can inspect the correct storage metrics once it fast syncs, right?
We can send a call to ask for the available history range? (can we? how? :P @JoshLind )

File: storage/aptosdb/src/lib.rs (hunk 2 above)
Comment (grao1991) [reply storage/aptosdb/src/lib.rs:1915]: This is the only place :)

We can wrap them later if necessary.

**Comment by grao1991** [comment]
> > whether or not you can inspect the correct storage metrics once it fast syncs, right?
> 
> We can send a call to ask for the available history range? (can we? how? :P @JoshLind )

I've added some asserts in smoke test to verify the metrics, PTAL.

File: testsuite/smoke-test/src/state_sync.rs [hunk 3]
```diff
@@ -50,6 +50,27 @@ async fn test_full_node_bootstrap_state_snapshot() {
...
+        .unwrap();
+    let ledger_pruner_version = inspection_client
+        .get_node_metric_i64("aptos_pruner_min_readable_version{pruner_name=ledger_pruner}")
+        .await
+        .unwrap()
+        .unwrap();
+
+    assert!(state_merkle_pruner_version > 0);
+    assert!(epoch_snapshot_pruner_version > 0);
+    assert!(ledger_pruner_version > 0);
```
Comment (msmouse) [inline testsuite/smoke-test/src/state_sync.rs:73]: 👍

**Review by msmouse — APPROVED** [review]

**Pull request merged**